### DISCUSSION TITLE
Getting CI builds working for Mac and Windows again

### DISCUSSION THREAD

**Discussion by pyjamasam** [collin80/SavvyCAN#352] [General]
I have had builds working for Linux, MacOS and Windows working on my fork using GitHub actions and I was wondering if I could assist with getting the builds working for the main repo.

Whats the best way for me to start helping with this?

I can submit a pull request with the build.yml that works for github if you'd like.  Or do you need it adapted for travis?

chris.

**Comment by collin80** [comment]
Currently I'm using TravisCI for Linux and AppVeyor for Windows. I used to have OSX working in TravisCI but it broke and I could not fix it. If you've got it all working with GitHub actions that'd be wonderful. I'm not necessarily beholden to either TravisCI nor AppVeyor. To be honest, I had NO idea what I was doing and both are working simply by trial and error. Both are kind of a mess. If you have it all working in GitHub actions I'm perfectly willing to switch to that. So, yes, if you could submit a pull request that implements this I'll test it and once its up and running I'll say goodbye to the two current systems I'm using to do the CI/CD builds.

**Reply by pyjamasam** [reply]
I just submitted a PR for this (#356).

It doesn't do macos signing, but it should be possible to extend and add it.
There is also a commented out section at the bottom for notifying a discord channel if you want.

Let me know if you have questions or need any help with any of this.

**Reply by collin80** [reply]
I merged your pull request but it seemed like it didn't recognize it. Some research suggested that the directory should be named workflows not workflow. So, I made a simple commit to rename the directory. This seems to cause the actions tab on github to recognize it now. We'll see if anything happens.

**Reply by pyjamasam** [reply]
Ahh my apologies.  I was fighting with it in my fork and was working in the master branch cause it wasn't working in my branch, and I suspect I didn't notice that the directory name was wrong.  Doah.  Sorry about that.

Ya.  I see that a build worked and everything was successful.  Thats sweet.  There are ways to get it to automatically go as well when you tag or what not.  I have it doing that for my CANServer repo when I get releases ready.  It's handy.

**Reply by collin80** [reply]
Yes, the build worked. I have suspended both TravisCI and Appveyor so that now I should get just GitHub Actions running. It doesn't look like it currently places the built files into the "Releases" but I'm sure it could.

**Reply by pyjamasam** [reply]
Right now all the build does is connects up the produced binaries as artifacts.  So they can be downloaded from the individual builds.

If you want to deploy as a release to the repo you can add a block to the end (similar to where the commented out discord notification is that would do that.

It looks like the official github release action is deprecated https://github.com/actions/create-release, but there is a list of other actions that are being maintained that could do it.

**Reply by collin80** [reply]
After a lot of failures... quite publicly in the master branch (whoops...) I did get it more or less working for automatic CI builds that show up under releases. The Windows build seems suspiciously small, probably because the VS runtime isn't in there. I probably should add the help files too. But, it's getting there.

**Reply by pyjamasam** [reply]
It's looking good.  I pulled the mac build and it seems great!
Let me know if i can help at all with any of the remaining stuff you want to do.
